src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs

```diff
@@ -1276,5 +1297,203 @@ private bool TryTranslateSumResultOperator(PipelineExpression node, out BsonValu
+            {
+                element = new BsonElement();
+                var firstLetterIndex = value.IndexOf(MapTemplateStartDelimiter, StringComparison.Ordinal);
+                if (firstLetterIndex == -1)
+                {
+                    return false;
+                }
+
+                firstLetterIndex += MapTemplateStartDelimiter.Length;
+                var lastLetterIndex = value.IndexOf(MapTemplateEndDelimiter, StringComparison.Ordinal);
+                if (lastLetterIndex == -1)
+                {
+                    return false;
+                }
+
+                if (lastLetterIndex > firstLetterIndex)
+                {
+                    var elementValue = value.Substring(firstLetterIndex, lastLetterIndex - firstLetterIndex);
+                    element = BsonDocument.Parse(elementValue).GetElement(0);
+                    return true;
+                }
+
+                return false;
+            }
+
+            private static string AddTemplateDelimiterBrackets(string value)
+            {
+                return $"{MapTemplateStartDelimiter}{value}{MapTemplateEndDelimiter}";
+            }
+
+            private static string RemoveMapPrefixIfThere(string value)
+            {
+                if (!string.IsNullOrWhiteSpace(value))
+                {
+                    var dotIndex = value.IndexOf('.');
+                    if (dotIndex != -1)
+                    {
+                        return $"{value.Substring(dotIndex + 1)}";
+                    }
+                }
+                return value;
+            }
+
+            private static Dictionary<string, List<string>> CollectTemplates(string value)
+            {
+                var templates = new Dictionary<string, List<string>>(); ;
+                int startIndex = 0;
+                while (startIndex != -1)
+                {
+                    string template;
+                    if (TryParseTemplate(value, ref startIndex, out template))
+                    {
+                        BsonElement element;
+                        if (TryGetTemplateElement(template, out element))
+                        {
+                            if (templates.ContainsKey(element.Name))
+                            {
+                                templates[element.Name].Add(element.Value.ToString());
+                            }
+                            else
+                            {
+                                templates.Add(element.Name, new List<string> {element.Value.ToString()});
+                            }
+                        }
+                        else
+                        {
+                            throw new MongoException("The query transformation has not been finished: Incorrect template format.");
+                        }
+                    }
+                }
+
+                return templates;
+            }
+
+            private static string CreateDocument(string ket, string value)
```

Comment:
key?

Reply:
Sorry, I've fixed this. Just pushed not last version :(

Reply:
Last version has been already pushed.